src/_guides/language/effective-dart/design.md

```diff
@@ -1151,6 +1212,90 @@ for one or more of the function's parameters. In those cases, you may need to
 annotate.
 
 
+### AVOID type annotating generic invocations that can be inferred.
+
+It's often useful to annotate *declarations* because they form the boundary of your API. But when instantiating an object or invoking a generic method, the type is often just noise. If inference would produce the correct type, omit the annotation and let it do it's job.
+
+When this rule is followed consistently, it helps draw the reader's attention to cases where inference does *not* produce the right type. The presence of the type argument sends a signal that that type is not otherwise pinned down by the surrounding code, and is telling you something useful.
+
+{:.good-style}
+<?code-excerpt "misc/lib/effective_dart/design_good.dart (generic-invocation)"?>
+{% prettify dart %}
+var strings = ["str", "ing"];
+var threes = new List.filled(3, 2);
+var pointThrees = threes.map((i) => i * 0.1);
+{% endprettify %}
+
+{:.bad-style}
+<?code-excerpt "misc/lib/effective_dart/design_bad.dart (generic-invocation)"?>
+{% prettify dart %}
+var strings = <String>["str", "ing"];
+var threes = new List<int>.filled(3, 2);
+var pointThrees = threes.map<double>((i) => i * 0.1);
+{% endprettify %}
+
+
+### PREFER type annotating declarations for top-level variables and fields that can't be inferred.
+
+Sometimes the initializer expression doesn't contain enough information to fully
+infer the type of a declaration and you need to give it a nudge. You often have
+two options:
+
+*   You can annotate the declaration on the left, which inference then pushes
+    over to the generic invocation on the right:
+
+    {:.good-style}
+    <?code-excerpt "misc/lib/effective_dart/design_good.dart (explicit-field)"?>
+    {% prettify dart %}
+    class Histogram {
+      final Map<String, int> counts = {};
+    }
+    {% endprettify %}
+
+    (We call this "downwards inference" because we think of the code as a tree
+    and the initializer expression is a subtree of the larger declaration tree.)
+
+*   You can annotate the generic invocation expression in the initializer, which
+    inference then uses to infer the field or variable's type:
+
+    {:.bad-style}
+    <?code-excerpt "misc/lib/effective_dart/design_bad.dart (explicit-field)"?>
+    {% prettify dart %}
+    class Histogram {
+      final counts = <String, int>{};
+    }
+    {% endprettify %}
+
+    (Conversely, we call this "upwards inference" because the type is flowing up
+    from the initializer expression to the surrounding declaration.)
+
```

Comment:
Nit: the list text is over indented.

Reply:
You need to do +4 for Markdown to recognize a subsequent paragraph as part of the previous bullet list.

Reply:
To be clearer: the start of the list is over indented currently as

    *   You can ...

It should be

    * You can ...

That will allow you to unindent the lines that follow.

Reply:
(By not over-indentation you can avoid situations where some of you markdown is interpreted as a code block, like in https://github.com/dart-lang/site-www/pull/819/files/359fa138fd682b184f4ca2d289f4e7ed9e3c2bff#r185097368)